website/project/model.py

```diff
@@ -3478,6 +3420,236 @@ def require_approval(self, user, notify_initiator_on_complete=False):
+                    }
+                },
+                auth=Auth(user)
+            )
+        return True
+
+    def remove_primary_institution(self, user, log=True):
+        inst = self.primary_institution
+        if not inst:
+            return False
+        self.primary_institution = None
+        if inst in self.affiliated_institutions:
+            self.affiliated_institutions.remove(inst)
+        if log:
+            self.add_log(
+                action=NodeLog.PRIMARY_INSTITUTION_REMOVED,
+                params={
+                    'node': self._primary_key,
+                    'institution': {
+                        'id': inst._id,
+                        'name': inst.name
+                    }
+                },
+                auth=Auth(user)
+            )
+        return True
+
+    def institution_url(self):
+        return self.absolute_api_v2_url + 'institution/'
+
+    def institution_relationship_url(self):
+        return self.absolute_api_v2_url + 'relationships/institution/'
+
+
+class SpecialList(list):
+    def __init__(self, node, private_target, init=None):
+        super(SpecialList, self).__init__(init or [])
+        self.node = node
+        self.target = private_target
+
+    def append(self, to_append):
+        junk = getattr(self.node, self.target)
+        junk.append(to_append.node)
+        setattr(self.node, self.target, junk)
+
+    def remove(self, to_remove):
+        junk = getattr(self.node, self.target)
+        junk.remove(to_remove.node)
+        setattr(self.node, self.target, junk)
+
+
+class InstitutionQuerySet(MongoQuerySet):
+    def __init__(self, queryset):
+        super(InstitutionQuerySet, self).__init__(queryset.schema, queryset.data)
+
+    def __iter__(self):
+        for each in super(InstitutionQuerySet, self).__iter__():
+            yield Institution(each)
+
+    def _do_getitem(self, index):
+        item = super(InstitutionQuerySet, self)._do_getitem(index)
+        if isinstance(item, MongoQuerySet):
+            return self.__class__(item)
+        return Institution(item)
+
+class Institution():
+    '''
+    "wrapper" class for Node. Together with the find and institution attributes & methods in Node,
+    this is to be used to allow interaction with Institutions, which are Nodes (with ' is_institution ' == True),
+    as if they were a wholly separate collection. To find an institution, use the find methods here,
+    and to use a Node as Institution, instantiate an Institution with ' Institution(node) '
+    '''
+    attribute_map = {
+        '_id': 'institution_id',
+        'auth_url': 'institution_auth_url',
+        'domain': 'institution_domain',
+        'name': 'title',
+        'logo_name': 'institution_logo_name',
+        'description': 'description',
+        'email_domain': 'institution_email_domain'
+    }
+
+    def __init__(self, node):
+        if node is None:
+            return
```

Comment:
This isn't entirely intuitive, because it will result in instances that may or may not have the `node` attribute.

```python
institution = Institution(node)
institution.node # => Node

institution = Institution()
institution.node # => AttributeError
```

Instead, I would expect:

```python
institution = Institution()
institution.node # => None
```